src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1450,23 +1450,41 @@ class LedgerMasterImp
         if (ledgerHash)
             return *ledgerHash;
 
-        // No, Try to get another ledger that might have the hash we need
-        // Compute the index and hash of a ledger that will have the hash we need
-        LedgerIndex refIndex = (index + 255) & (~255);
+        // The hash is not in the reference ledger. Get another ledger which can
+        // be located easily and should contain the hash.
```

Comment:
"Get the next page header and see if it holds the ledger." or something like that.